spec/lib/task-spec.js

```diff
@@ -508,6 +502,64 @@ describe("Task", function () {
 
     });
 
+    describe("timer", function() {
+        it("should not timeout when value is 0", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = 0;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should not timeout when value is -1", function() {
+            var definition = _.cloneDeep(noopDefinition);
+            definition.options.timeout = -1;
+            var task = Task.create(definition, {}, {});
+            task.renderAll = sinon.stub().resolves();
+            sinon.spy(task, 'cancel');
+
+            return task.run().then(function() {
+                expect(task.state).to.equal('succeeded');
+                expect(task.error).to.equal(null);
+            });
+        });
+
+        it("should timeout with options.timeout", function() {
```

Comment:
Sorry, I don't understand why this test case works well, because I don't see any Promise.delay, so how the timeout happens?
@benbp : could you please give some explanation.